server/src/main/java/org/jetbrains/bsp/bazel/server/bloop/BloopExporter.java

```diff
@@ -33,6 +34,19 @@ class BloopExporter {
   private final WorkspaceContextProvider workspaceContextProvider;
   private final Path workspaceRoot;
 
+  static void validateNoFailedExternalTargets(
+      Set<BuildTargetIdentifier> projectTargets, Set<BuildTargetIdentifier> failedTargets)
+      throws BazelExportFailedException {
+    var failedExternalTargets =
```

Comment:
it works for me, but it could be done with `.removeAll` - just calculate the difference and check is it empty or not

Reply:
yeah, I do prefer treating things as immutable though, particularly mutating input parameters.